assets/js/timezone.ts

```diff
@@ -0,0 +1,57 @@
+export function localTimezoneName() {
+  return Intl.DateTimeFormat().resolvedOptions().timeZone;
+}
+
+/**
+ * @param utcTime string of time in UTC such as "5pm", "6am", "6:30am"
+ * @return converted to local time such as "8pm", "2:30pm"
+ */
+export function convertToLocal(utcTime: string) {
+  const match = utcTime.match(/(\d+)(:(\d+))?\s*([ap].?m)/i);
+
+  if (!match) {
+    return "N/A";
+  }
+
+  const [, hourStr, , minuteStr, ampm] = match;
+
+  const ampmOffset = ampm.toLowerCase().startsWith("p") ? 12 : 0;
+
+  const localDate = new Date(Date.now());
+  localDate.setUTCHours(parseInt(hourStr, 10) + ampmOffset);
+  if (minuteStr) {
+    localDate.setUTCMinutes(parseInt(minuteStr, 10));
+  } else {
+    localDate.setUTCMinutes(0);
+  }
+
+  // Comes back like "7:01 PM"
+  const formatted = new Intl.DateTimeFormat(undefined, {
+    hour: "numeric",
+    minute: "2-digit",
+    hour12: true,
+  }).format(localDate);
+
+  return formatted.toLowerCase().replace(" ", "").replace(":00", "");
```

Comment:
Wondering about the first argument (locale) of `DateTimeFormat` and the `hour12` option:

- By specifying `undefined`, we're relying on the user's locale, and it doesn't seem like we should hard-control the 12-hour setting, and our string replacements may not be totally accurate.
- Alternatively, we could force it to `'en-US'` which makes it more predictable (and probably still don't need `hour12`), and would be consistent with the rest of the page otherwise being in English

```suggestion
  const formatted = new Intl.DateTimeFormat('en-US', {
    hour: "numeric",
    minute: "2-digit",
  }).format(localDate);

  return formatted.toLowerCase().replace(" ", "").replace(":00", "");
```